### ISSUE TITLE
Parsing POTCAR: 'utf8' codec can't decode byte 0x97 in position 37: invalid start byte

### ISSUE THREAD

**Issue by manomars** [materialsproject/pymatgen#686]
## System

* Pymatgen version: v4.7.7
* Python version: 2.7.12
* OS version: Linux Ubuntu 14.04

## Summary

* Trying to parse POTCAR I get:
UnicodeDecodeError: 'utf8' codec can't decode byte 0x97 in position 37: invalid start byte

## Example code

```python
from pymatgen.io.vasp import Vasprun
v = Vasprun("./vasprun.xml")
print(v.final_energy)
```

## Error message
/fs/home/mmars/matgen/local/lib/python2.7/site-packages/pymatgen/io/vasp/outputs.py in __init__(self, filename, ionic_step_skip, ionic_step_offset, parse_dos, parse_eigen, parse_projected_eigen, parse_potcar_file, occu_tol, exception_on_bad_xml)
    381 
    382             if parse_potcar_file:
--> 383                 self.update_potcar_spec(parse_potcar_file)
    384 
    385         if not self.converged:

/fs/home/mmars/matgen/local/lib/python2.7/site-packages/pymatgen/io/vasp/outputs.py in update_potcar_spec(self, path)
    827                 potcar = get_potcar_in_path(path)
    828         elif isinstance(path, bool) and path:
--> 829             potcar = get_potcar_in_path(os.path.split(self.filename)[0])
    830         else:
    831             potcar = None

/fs/home/mmars/matgen/local/lib/python2.7/site-packages/pymatgen/io/vasp/outputs.py in get_potcar_in_path(p)
    811             for fn in os.listdir(os.path.abspath(p)):
    812                 if 'POTCAR' in fn:
--> 813                     pc = Potcar.from_file(os.path.join(p, fn))
    814                     if {d.header for d in pc} == \
    815                             {sym for sym in self.potcar_symbols}:

/fs/home/mmars/matgen/local/lib/python2.7/site-packages/pymatgen/io/vasp/inputs.pyc in from_file(filename)
   1711     def from_file(filename):
   1712         with zopen(filename, "rt") as reader:
-> 1713             fdata = reader.read()
   1714         potcar = Potcar()
   1715         potcar_strings = re.compile(r"\n?(\s*.*?End of Dataset)",

/fs/home/mmars/matgen/lib/python2.7/codecs.pyc in decode(self, input, final)
    312         # decode input (taking the buffer into account)
    313         data = self.buffer + input
--> 314         (result, consumed) = self._buffer_decode(data, self.errors, final)
    315         # keep undecoded input until the next call
    316         self.buffer = data[consumed:]

UnicodeDecodeError: 'utf8' codec can't decode byte 0x97 in position 37: invalid start byte

**Comment by shyuep** [comment]
Can you please provide the POTCAR that is being parsed?

**Comment by shyuep** [comment]
I just pushed a changed which will try to ignore Unicode decoding errors in Potcar. Can you pull the latest changes and see if it solves your problems?

**Comment by mkhorton** [comment]
@shyuep regarding your changes, is there an advantage in using `codecs` over `zopen(filename, "rt", errors="replace")`? I ran into a similar issue in cif.py with unknown characters, usually in author names

**Comment by shyuep** [comment]
@mkhorton I am not sure if there are any advantages per se. I would say ignoring errors probably result in a higher likelihood of success. In general, I am not too worried about POTCAR parsing since the information in the POTCAR is seldom used for final analysis (except in very specialized cases such as counting electrons, etc.). So I would bias the code against a much successful parse rather than an absolutely accurate parse.

I am not able to reproduce the same error either in Py36 or Py27 using all VASP PSPs from the old 2012 version all the way to the newest 54 version. Cycling through all the PSPs resulted in a successful read even before I made the codec change, though there are some PSPs, especially the H PSPs that did not have description and other fields which caused a failed read. I have fixed those.

**Comment by manomars** [comment]
Hi,

I tracked down the problem:
In the directory where my vasprun.xml and POTCAR file reside there was a file called ._POTCAR present.
This useless file is created by this horrible Mac Finder. Probably this happened when I mounted my partition on our fileserver on my macbook at some point …
Anyway: the pymatgen input scripts where trying to read from ._POTCAR instead of POTCAR.
I removed all ._* files and now everything runs without trouble (without you latest change as well).

Sorry to have waisted your time!
Many thanks for your efforts!

Cheers,
Martijn
